include/constants.h

```diff
@@ -74,5 +74,16 @@ namespace reinforcement_learning {  namespace value {
       const bool DEFAULT_MODEL_BACKGROUND_REFRESH = true;
       const int DEFAULT_VW_POOL_INIT_SIZE = 4;
       const int DEFAULT_PROTOCOL_VERSION = 1;
+#ifdef USE_AZURE_FACTORIES
```

Comment:
Having a define in a public header that doesn't depend on the compiler/system is a bad practice and we should avoid if possible.

They should be functions and the ifdef should exist within their implementation.

I don't understand the change to the default clock.

Reply:
Thanks for your feedback, @kumpera! That being the case, should we define static functions and implement them in a new constants.cc?

As far as the change to the default clock, it was my understanding that the file based logger requires a time provider implementation. Since `NULL_TIME_PROVIDER` was the current default, that would cause an issue. @lokitoth is that right?